provider/pkg/naming/convert.go

```diff
@@ -288,6 +288,47 @@ func cfnValueToSdk(value interface{}) interface{} {
 	}
 }
 
+// ToStringifiedMap creates a copy of the input map with all primitive values converted to strings.
+func ToStringifiedMap(value map[string]interface{}) map[string]interface{} {
+	if value == nil {
+		return nil
+	}
+
+	result := map[string]interface{}{}
+	for k, v := range value {
+		result[k] = primitivesToString(v)
+	}
+	return result
+}
+
+func primitivesToString(value interface{}) interface{} {
+	if value == nil {
+		return nil
+	}
+
+	switch reflect.TypeOf(value).Kind() {
+	case reflect.Map:
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
```

Comment:
Under what circumstance would this happen (`!ok`)? Should we throw an error here instead?

Reply:
In this case it shouldn't happen because CloudFormation only supports strings as keys.

Throwing sounds more apt in this case. Will change it accordingly!